ui/webui/src/components/storage/InstallationDestination.jsx

```diff
@@ -118,13 +119,13 @@ const LocalStandardDisks = () => {
             </CardHeader>
             <CardBody>
                 <DataList isCompact aria-label={_("Usable disks")}>
-                    {usableDisks && usableDisks.map(disk => (
+                    {Object.keys(disks).map(disk => (
                         <DataListItem key={disk} aria-labelledby={"local-disks-checkbox-" + disk}>
                             <DataListItemRow>
                                 <DataListCheck
                                   aria-labelledby={"local-disks-checkbox-" + disk}
-                                  onChange={value => setSelectedDisks({ ...selectedDisks, [disk]: value })}
-                                  checked={!!selectedDisks[disk]}
+                                  onChange={value => setDisks({ ...disks, [disk]: value })}
```

Comment:
These two lines look like quite some magic! :) What does it do actually ?

Reply:
``` js
value => setDisks({ ...disks, [disk]: value })

```
can be written without arrow functions syntax as:

``` js
function (value) {
   setDisks({ ...disks, [disk]: value });
}
```
And if avoid even using the object destructuring it can be written as:

```js
function (value) {
  // We want to create a new object, by using the object 'disks' and adding the new disk
  // 'disks' object is part of the state, so we should not update it directly
  // Instead we need to update the state with setDisks

  // Create a new clone of disks
  const newDisks = Object.assign({},disks);
  // Add the new disk value
  newDisks[disk] = value; 
  // Update the state
  setDisks(newDisks);
}
```

The previous can be so much nicer be done with the spread operator, to avoid creating a new variable.

Reply:
Does this make sense? If not let's talk it over in a call.

Reply:
Just to mention ths once more, the reason for this complexity here, is that we really * have to * update the state in an immutable way. Mutatiing state is a big NO in react ecosystem, so any operation here updating the 'disks' state object directly is not allowed.